File: introduccion/Cuentas_Bancarias/src/main/java/com/example/cuentas_bancarias/Model/CuentaAhorro.java
package com.example.cuentas_bancarias.Model;

import com.example.cuentas_bancarias.Interface.Imprimible;

public class CuentaAhorro extends CuentaBancaria implements Imprimible {
    private double tipoInteres;

    public CuentaAhorro(Persona titularCuenta, double saldoActualCuenta, String numCuenta, double tipoInteres) {
        super(titularCuenta, saldoActualCuenta, numCuenta);
        this.tipoInteres = tipoInteres;
    }

    public double getTipoInteres() {
        return tipoInteres;
    }
    public void setTipoInteres(double tipoInteres) {
        this.tipoInteres = tipoInteres;
    }

    @Override
    public String devolverInfoString() {
        // SE IMPLEMENTA EL METODO DE LA CLASE ABSTRACTA Y SE AÑADE INFORMACION
        return "Titular: " + getTitularCuenta().devolverInfoString() +  // INFO TITULAR
                "\nSaldo actual: " + getSaldoActualCuenta() +  // SACDO ACTUAL
                "\nNúmero de cuenta: " + getNumCuenta() +  // NUMERO DE CUENTA
                "\nTipo de cuenta: Ahorro" +  // TIPO DE CUENTA
                "\nInterés de remuneración: " + getTipoInteres();  // INTERES
    }
}

File: Tema2_Kotlin/Ejercicios/CuentasBancariasKt/src/main/kotlin/Model/CuentaAhorro.kt
package Model

class CuentaAhorro(
    titular: Persona,
    saldo: Double,
    iban: String,
    private var tipoInteres: Double
) : CuentaBancaria(titular, saldo, iban) {

    override fun devolverInfo(): String {
        return """
        Cuenta Ahorro:
        -----------------
        Titular         : $titular
        Saldo           : $saldo
        IBAN            : $iban
        Tipo Interes : $tipoInteres
    """.trimIndent()
    }
}